bids/layout/models.py

```diff
@@ -702,17 +704,11 @@ def __repr__(self):
 
     @reconstructor
     def _init_on_load(self):
-        if self._dtype not in ('str', 'float', 'int', 'bool', 'json'):
-            raise ValueError("Invalid dtype '{}'. Must be one of 'int', "
-                             "'float', 'bool', 'str', or 'json'.".format(self._dtype))
         if self._dtype == 'json':
-            self.value = json.loads(self._value)
+            self.value = self._value
```

Comment:
I suspect this needs reverting.
```suggestion
            self.value = json.loads(self._value)
```